framework/src/wifi_manager/wifi_manager.c

```diff
@@ -149,337 +319,268 @@ static wifi_utils_result_e stop_dhcp_server(void)
+		ndbg("handle connect_event fail\n");
 	}
-	wifi_mutex_release(w_mutex);
+}
 
-	wifi_mutex_acquire(w_info_mutex, WIFI_UTILS_FOREVER);
-	strncpy(g_manager_info.ssid, config->ssid, config->ssid_length + 1);
-	wifi_mutex_release(w_info_mutex);
 
-	return WIFI_MANAGER_SUCCESS;
+void _wifi_utils_join_event(void *arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_JOIN, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle connect_event fail\n");
+	}
 }
 
-wifi_manager_result_e wifi_manager_disconnect_ap(void)
-{
-	CHECK_WITH_RETURN("disconnect_ap");
 
-	wifi_utils_result_e result = WIFI_UTILS_SUCCESS;
+void _wifi_utils_leave_event(void * arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_LEAVE, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle connect_event fail\n");
+	}
+}
 
-	wifi_mutex_acquire(w_mutex, WIFI_UTILS_FOREVER);
-	result = wifi_utils_disconnect_ap();
-	wifi_mutex_release(w_mutex);
 
-#ifdef CONFIG_ENABLE_IOTIVITY
-	if (result == WIFI_UTILS_SUCCESS) {
-		__tizenrt_manual_linkset("del");
+void _wifi_utils_scan_done(wifi_utils_result_e result, wifi_utils_scan_list_s *slist, void *arg)
+{
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_SCAN_DONE, (void *)slist};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle scan done event fail\n");
 	}
-#endif
-
-	return result;
 }
 
-wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
+void _wifi_dhcpd_event(void)
 {
-	wifi_utils_result_e result = WIFI_UTILS_SUCCESS;
-	wifi_utils_info info;
-
-	if (wmcb == NULL) {
-			ndbg("Wi-Fi Manager init fail because of no callbacks");
-			return WIFI_MANAGER_FAIL;
+	WM_LOG_START;
+	_wifimng_msg_s msg = {EVT_DHCPD_GET_IP, NULL};
+	wifi_manager_result_e res = _handle_request(&msg);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		ndbg("handle dhcpd_event fail\n");
 	}
 
-	if (w_info_mutex == NULL) {
-		w_info_mutex = (wifi_mutex *)malloc(sizeof(wifi_mutex));
-		if (w_info_mutex == NULL) {
-			ndbg("w_info_mutex malloc fails.\n");
-			return WIFI_MANAGER_FAIL;
-		}
-		result = wifi_mutex_create(w_info_mutex);
-		if (result != WIFI_UTILS_SUCCESS) {
-			ndbg("w_info_mutex_create fail");
-			goto error_without_mutex_release;
-		}
-	}
+	return ;
+}
 
-	wifi_mutex_acquire(w_info_mutex, WIFI_UTILS_FOREVER);
-	if (w_mutex != NULL) {
-		ndbg("WI-FI is already initialized.\n");
-		wifi_mutex_release(w_info_mutex);
-		return WIFI_MANAGER_INITIALIZED;
+/*
+ * State Functions
+ */
+wifi_manager_result_e _handle_init_state(_wifimng_msg_s *msg)
+{
+	WM_LOG_HANDLER_START;
+	_wifimng_evt_e evt = msg->event;
+	if (evt != EVT_INIT) {
+		return WIFI_MANAGER_FAIL;
 	}
 
-	w_mutex = (wifi_mutex *)malloc(sizeof(wifi_mutex));
-	if (w_mutex == NULL) {
-		ndbg("w_mutex malloc fails.\n");
-		wifi_mutex_release(w_info_mutex);
-		goto error_with_mutex_destory;
-	}
+	wifi_utils_result_e wres = wifi_utils_init();
```

Comment:
maybe check for wres, and return appropriately.

Reply:
I missed. I will add